src/exception.c

```diff
@@ -124,6 +167,10 @@ void print_regs(u64 *regs, int el12)
     u64 elr = el12 ? mrs(ELR_EL12) : mrs(ELR_EL1);
     u64 esr = el12 ? mrs(ESR_EL12) : mrs(ESR_EL1);
 
+    if (in_gl12()) {
```

Comment:
These are the extra guarded mode bits, right? Only available in GLx (no ASPSR_EL1 etc)? Sounds good to do it like this, but maybe put the print right after SPSR_EL1 since it extends it.

Reply:
Yes, that register only exists in guarded mode. I haven't found ASPSR_EL1 and I don't believe that it exists